From a90987b20a89f2a17b864ef59339ef5427bb99af Mon Sep 17 00:00:00 2001
From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
Date: Thu, 14 Sep 2023 15:53:20 +0530
Subject: [PATCH] Fix pip3 version error.

Pick pip version and set sleep inhibitor service.

Tracked-On: OAM-112234
Signed-off-by: Suresh, Prashanth <prashanth.suresh@intel.com>
---
 scripts/setup_host.sh | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/scripts/setup_host.sh b/scripts/setup_host.sh
index e800d10..8296955 100755
--- a/scripts/setup_host.sh
+++ b/scripts/setup_host.sh
@@ -414,7 +414,10 @@ function ubu_update_bt_fw() {
 function set_sleep_inhibitor() {
     sudo apt-get -y install python3-pip
     sudo pip3 install -U sleep-inhibitor
-    sudo sed -i 's/\/usr\/bin\/%p/\/usr\/local\/bin\/%p/' /usr/local/share/sleep-inhibitor/sleep-inhibitor.service
+
+    pythonversion="$(pip3 --version | grep -Po '^.*\(\K[^\)]*' | grep -Po '^.*\ \K[^\\n]*')"
+    echo "pythonversion = $pythonversion"
+    sudo sed -i 's/\/usr\/bin\/%p/\/usr\/local\/bin\/%p/' /usr/local/lib/python$pythonversion/dist-packages/sleep_inhibitor/sleep-inhibitor.service
     #Download the plugin if not already
     sudo echo "#! /bin/sh
 if adb get-state 1>/dev/null 2>&1
@@ -427,9 +430,9 @@ then
         fi
 else
         exit 0
-fi" > /usr/local/share/sleep-inhibitor/plugins/is-wakelock-active
-    sudo chmod a+x /usr/local/share/sleep-inhibitor/plugins/is-wakelock-active
-    sudo cp /usr/local/share/sleep-inhibitor/sleep-inhibitor.conf /etc/.
+fi" > /usr/local/lib/python$pythonversion/dist-packages/sleep_inhibitor/plugins/is-wakelock-active
+    sudo chmod a+x /usr/local/lib/python$pythonversion/dist-packages/sleep_inhibitor/plugins/is-wakelock-active
+    sudo cp /usr/local/lib/python$pythonversion/dist-packages/sleep_inhibitor/sleep-inhibitor.conf /etc/.
     sudo echo "plugins:
     #Inhibit sleep if wakelock is held
     - path: is-wakelock-active
@@ -437,7 +440,7 @@ fi" > /usr/local/share/sleep-inhibitor/plugins/is-wakelock-active
       what: sleep
       period: 0.01" > /etc/sleep-inhibitor.conf
     sudo sed -i 's/#*HandleSuspendKey=\w*/HandleSuspendKey=suspend/' /etc/systemd/logind.conf
-    sudo cp /usr/local/share/sleep-inhibitor/sleep-inhibitor.service /etc/systemd/system/.
+    sudo cp /usr/local/lib/python$pythonversion/dist-packages/sleep_inhibitor/sleep-inhibitor.service /etc/systemd/system/.
     reboot_required=1
 }
 
-- 
2.42.0

